distributed/executor.py

```diff
@@ -976,6 +987,66 @@ def cancel(self, futures):
         return sync(self.loop, self._cancel, futures)
 
     @gen.coroutine
+    def _publish_dataset(self, data, name):
+        keys = [tokey(f.key) for f in futures_of(data)]
+        yield self.scheduler.publish_dataset(keys=keys, name=tokey(name),
+                                             data=pickle.dumps(data),
+                                             client=self.id)
+
+
+    def publish_dataset(self, data, name):
+        """
+        Store a named reference to the data in the scheduler for other executors
+
+        Parameters
+        ----------
+        data: list of futures or dask collection
+        name: str or container
+            handle by which the stored data should be known. If not a string,
+            will be wrapped by tokey() to deterministically turn into one.
+
+        Returns
+        -------
+        None
+        """
+        sync(self.loop, self._publish_dataset, data, name)
+
+    @gen.coroutine
+    def _unpublish_dataset(self, name):
+        yield self.scheduler.unpublish_dataset(name=name)
```

Comment:
Slight preference to remove this one-line coroutine and instead inline `self.scheduler.unpublish_dataset` wherever it gets used.